lib/helpers.ts

```diff
@@ -137,6 +137,17 @@ export function fillInEntities(data: LoadDataResult): Entity[] {
     const diagnosisMap = data.diagnosisByHTANParticipantID;
     const demoMap = data.demographicsByHTANParticipantID;
 
+    // give each biospecimen it's caseid (i.e "diagnosis" HTANParticipantID)
+    // biospecimen have HTANParentID but that may or may not be it's caseid because
+    // biospecimen can have other biospecimen as parents (one case at top)
+    _.forEach(data.biospecimenByHTANBiospecimenID, (specimen) => {
+        const parentIdMatch = specimen.HTANParentID.match(/[^_]*_[^_]*/);
+        // we should always have a match
+        specimen.HTANParticipantID =
+            specimen.HTANParticipantID ||
+            (parentIdMatch ? parentIdMatch[0] : '');
+    });
```

Comment:
it seems like similar logic lives here:

https://github.com/ncihtan/htan-portal/blob/f2c54b71dda65cb620b4274589659434a4842a78/data/processSynapseJSON.ts#L272-L274

Not sure if we can unify this somehow

Reply:
filed an issue for this https://github.com/ncihtan/htan-portal/issues/367